bluemira/mesh/meshing.py

```diff
@@ -0,0 +1,721 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+Core functionality for the bluemira mesh module.
+"""
+
+from __future__ import annotations
+
+import copy
+import pprint
+from typing import TYPE_CHECKING, Dict, Union
+
+# import mesher lib (gmsh)
+import gmsh
+
+# import bluemira modules
+import bluemira.geometry as geo
+
+from .error import MeshOptionsError
+
+if TYPE_CHECKING:
+    from bluemira.base.components import Component
```

Comment:
@DanShort12, @je-cook, I am trying to solve a cyclic import problem using type checking (in the same way one of you did in another module with BluemiraGeo). However, when I run the example/mesh/mesh_tutorial.py, I get `NameError: name 'Component' is not defined`. Any idea on how to solve this problem?

Reply:
`TYPE_CHECKING` only protects against circular imports on annotated types (eg `myvar: Union[MyFancyClass, None] = None`) in function/class definitions. You would also need to `from __future__ import annotations` which converts all type annotations to strings therefore they do not need to be imported unless you are running something like `mypy` which at the moment we don't (we are just slowly heading that way)

It doesn't protect against direct type checking which it looks like you are doing